bin/assigner/src/main.cpp

```diff
@@ -825,8 +1205,9 @@ int main(int argc, char *argv[]) {
             return 1;
         }
     }
+    gen_mode = gen_mode | nil::blueprint::generation_mode::assignments(); // for now in any case we generate assignment table to determine used_rows
```

Comment:
we can move it to the line 1189

Reply:
gen mode must have assignments(), and it won't have if moved, code inside the "else if" will rewrite gen mode and remove assignments()